engine/src/main/java/org/hibernate/search/engine/metadata/impl/FacetMetadata.java

```diff
@@ -63,6 +70,7 @@ public FacetEncodingType getEncoding() {
 
 		// optional parameters
 		private FacetEncodingType encoding = FacetEncodingType.AUTO;
```

Comment:
Not related to your patch but I dislike the fact that the default value of encoding is AUTO while this value is not supported when exploiting `FacetMetadata` later on. It's quite misleading.

So defining the encoding type is mandatory, not optional.

Not sure what to do about it but I thought I might as well mention it.